src/main/java/org/osiam/resources/scim/Address.java

```diff
@@ -0,0 +1,397 @@
+    }
+
+    /**
+     * Constructor for serialization. It is not intended for public use.
+     */
+    private Address(Builder builder) {
+        super(builder);
+        this.formatted = builder.formatted;
+        this.streetAddress = builder.streetAddress;
+        this.locality = builder.locality;
+        this.region = builder.region;
+        this.postalCode = builder.postalCode;
+        this.country = builder.country;
+        this.type = builder.type;
+    }
+
+    /**
+     * Gets the full mailing address, formatted for display or use with a mailing label.
+     *
+     * @return the formatted address
+     */
+    public String getFormatted() {
+        return formatted;
+    }
+
+    /**
+     * Gets the full street address, which may include house number, street name, etc.
+     *
+     * @return the street address
+     */
+    public String getStreetAddress() {
+        return streetAddress;
+    }
+
+    /**
+     * Gets the city or locality
+     *
+     * @return the city or locality
+     */
+    public String getLocality() {
+        return locality;
+    }
+
+    /**
+     * Gets the state or region
+     *
+     * @return region the region
+     */
+    public String getRegion() {
+        return region;
+    }
+
+    /**
+     * Gets the postal code
+     *
+     * @return postalCode the postal code
+     */
+    public String getPostalCode() {
+        return postalCode;
+    }
+
+    /**
+     * Gets the country name in ISO 3166-1 alpha 2 format, e.g. "DE" or "US".
+     *
+     * @return the country
+     */
+    public String getCountry() {
+        return country;
+    }
+
+    /**
+     * Gets the type of the attribute.
+     * <p/>
+     * <p>
+     * For more detailed information please look at the <a href=
+     * "http://tools.ietf.org/html/draft-ietf-scim-core-schema-02#section-3.2" >SCIM core schema 2.0, section 3.2</a>
+     * </p>
+     *
+     * @return the actual type
+     */
+    public Type getType() {
+        return type;
+    }
+
+    @Override
+    public String getOperation() {
+        return super.getOperation();
+    }
+
+    @Override
+    public boolean isPrimary() {
+        return super.isPrimary();
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = super.hashCode();
+        result = prime * result + ((country == null) ? 0 : country.hashCode());
+        result = prime * result + ((formatted == null) ? 0 : formatted.hashCode());
+        result = prime * result + ((locality == null) ? 0 : locality.hashCode());
+        result = prime * result + ((postalCode == null) ? 0 : postalCode.hashCode());
+        result = prime * result + ((region == null) ? 0 : region.hashCode());
+        result = prime * result + ((streetAddress == null) ? 0 : streetAddress.hashCode());
+        result = prime * result + ((type == null) ? 0 : type.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method equals() has an NPath complexity of 131072](https://www.codacy.com/app/OSIAM/osiam/file/1979525052/issues/source?bid=3080997&fileBranchId=3117040#l185)